src/ipa_pc/mod.rs

```diff
@@ -896,9 +852,9 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             l_vec.push(lr[0]);
             r_vec.push(lr[1]);
 
-            round_challenge = Self::compute_random_oracle_challenge(
-                &to_bytes![round_challenge, lr[0], lr[1]].unwrap(),
-            );
+            fs_rng.absorb(&to_bytes![round_challenge, lr[0], lr[1]].unwrap());
+
+            round_challenge = fs_rng.squeeze_128_bits_challenge();
```

Comment:
No need to absorb the "current" challenge for squeezing the new one, as our Fiat Shamir rng is stateful.
Absorbing the LR's is good enough.

Reply:
Done